Framework/API/inc/MantidAPI/Algorithm.tcc

```diff
@@ -0,0 +1,144 @@
+#include "MantidAPI/Algorithm.h"
+#include "MantidAPI/IndexProperty.h"
+#include "MantidAPI/WorkspaceProperty.h"
+
+/* Currently required for Windows. Otherwise warning C4661 is issued. The
+ * ArrayProperty exlicit instantiations are not found.*/
+#ifdef _WIN32
```

Comment:
I think we need to somehow get to the bottom of this. If this is a compiler issue, we need to have a bug reference. Otherwise we will never know to be able to remove the preprocessor check.